pandas/src/algos_take_helper.pxi

```diff
@@ -4947,3 +4947,22 @@ def take_2d_multi_object_object(ndarray[object, ndim=2] values,
                     out[i, j] = fv
                 else:
                     out[i, j] = values[idx, idx1[j]]
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def take_bounds_check(int64_t[:] indexer, int64_t[:] out, int64_t n):
+    cdef:
```

Comment:
This is used by `maybe_convert_indices` - just faster (one pass) than what was being done before.  It became more of a bottleneck because it's used in more places.